### PR TITLE
Improve RMSProp documentation

### LINKED ISSUE #54368: RMSProp documentation is confusing
## 📚 Documentation

The documentation on the parameters for `torch.optim.RMSprop` is vague and seems to contradict itself. I couldn't tell what `alpha` means without looking at the code (it's the exponential discount factor for the second moment running average, which Keras calls `rho`). On the other hand, the sentence
> The effective learning rate is thus α/(sqrt(v) + ε) where α is the scheduled learning rate and v is the weighted moving average of the squared gradient.

uses alpha to mean the learning rate, which corresponds to `lr`. The documentation also does not explain how the second moment running average is computed, which would help explain how to use the discount rate `alpha`.

### PR DESCRIPTION
**Pull request by veeara282**
I've made the following changes to the documentation for `optim.RMSprop`:

- Rename the learning rate from alpha to _r_ to avoid confusion with the decay rate, which is named `alpha` in the RMSprop class.
- Call `alpha` the "decay rate."
- Describe how RMSprop updates the weighted moving average _v_ as a function of alpha, the current value of _v_, and the gradient _g_. (Notation due to [Prof. De Sa][desa21].) I've added this to the end, but we could move it to before the effective learning rate discussion if this is better.

Although the decay rate is often called rho, I've avoided renaming it to preserve backwards compatibility.

Fixes #54368

[desa21]: https://www.cs.cornell.edu/courses/cs4787/2021sp/notebooks/Slides9.html

### CODE REVIEW

**Comment by aidan-fitz** [comment]
Could someone please figure out why this PR is failing to build? @heitorschueroff

**Comment by iramazanli** [comment]
Thank you for this amazing PR, in a recent project we modified documentation of optimizers in a new style. That's why this PR is not relevant right now.
